Classes/Controller/SelectImageController.php

```diff
@@ -1,65 +1,87 @@
- * @subpackage Controller
- * @author     Christian Opitz <christian.opitz@netresearch.de>
- * @license    http://www.netresearch.de Netresearch Copyright
- * @link       http://www.netresearch.de
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
  */
 
+declare(strict_types=1);
 
 namespace Netresearch\RteCKEditorImage\Controller;
 
+use Exception;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use TYPO3\CMS\Core\Configuration\Richtext;
 use TYPO3\CMS\Core\Http\JsonResponse;
 use TYPO3\CMS\Core\Resource\File;
-use TYPO3\CMS\Core\Resource\Service\MagicImageService;
-use TYPO3\CMS\Core\Utility\GeneralUtility;
-use TYPO3\CMS\Recordlist\Controller\ElementBrowserController;
 use TYPO3\CMS\Core\Resource\ProcessedFile;
 use TYPO3\CMS\Core\Resource\ResourceFactory;
+use TYPO3\CMS\Core\Resource\Service\MagicImageService;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
 use TYPO3\CMS\Core\Utility\HttpUtility;
+use TYPO3\CMS\Recordlist\Controller\ElementBrowserController;
 
 /**
- * Controller for the image select wizard
- *
- * PHP version 7
+ * Controller for the image select wizard.
  *
- * @category   Netresearch
- * @package    RteCKEditor
- * @subpackage Controller
- * @author     Christian Opitz <christian.opitz@netresearch.de>
- * @license    http://www.gnu.de/documents/gpl-2.0.de.html GPL 2.0+
- * @link       http://www.netresearch.de
+ * @author  Christian Opitz <christian.opitz@netresearch.de>
+ * @license http://www.gnu.de/documents/gpl-2.0.de.html GPL 2.0+
+ * @link    http://www.netresearch.de
  */
 class SelectImageController extends ElementBrowserController
 {
+    /**
+     * @var bool
+     */
     protected bool $isInfoAction;
 
+    /**
+     * @var ResourceFactory
+     */
+    private ResourceFactory $resourceFactory;
+
+    /**
+     * @var Richtext
+     */
+    private Richtext $richText;
+
+    /**
+     * @var MagicImageService
+     */
+    private MagicImageService $magicImageService;
+
+    /**
+     * Constructor.
+     */
     public function __construct()
     {
+        // No dependency injection available here. :(
+        $this->resourceFactory = GeneralUtility::makeInstance(ResourceFactory::class);
+        $this->richText = GeneralUtility::makeInstance(Richtext::class);
+        $this->magicImageService = GeneralUtility::makeInstance(MagicImageService::class);
+
         $this->isInfoAction = GeneralUtility::_GP('action') === 'info';
+
         if (!$this->isInfoAction) {
             $bparams = explode('|', GeneralUtility::_GET('bparams'));
-            if (!$bparams[3]) {
+
+            if (isset($bparams[3]) && ($bparams[3] === '')) {
                 $bparams[3] = $GLOBALS['TYPO3_CONF_VARS']['GFX']['imagefile_ext'];
                 $_GET['bparams'] = implode('|', $bparams);
             }
         }
-        $this->mode = 'file';
+
+        $this->mode = GeneralUtility::_GET('mode') ?: 'file';
```

Comment:
I'm not really sure, may be I'm paranoid but this looks like an potential XSS Risk.

Reply:
I removed the assignment in commit #07769e4. The value is passed through the route to the parent, no need to assign it again.